src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new ArithmeticException("User name or password was incorrect");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
+        throw new ArithmeticException("User name or password was incorrect");
```

Comment:
can we simplify this?
<img width="736" alt="Screenshot 2021-12-14 at 09 45 55" src="https://user-images.githubusercontent.com/33026471/146001530-c1423785-040d-46f7-8951-dc35f9910b26.png">